lib/streamlit/web/server/routes.py

```diff
@@ -119,6 +120,11 @@ def validate_absolute_path(self, root: str, absolute_path: str) -> Optional[str]
 
         return super().validate_absolute_path(root, absolute_path)
 
+    def set_extra_headers(self, path: str) -> None:
+        if Path(path).suffix not in file_util.WHITELISTED_APP_STATIC_FILE_EXTENSIONS:
+            self.set_header("Content-Type", "text/plain")
+        self.set_header("X-Content-Type-Options", "nosniff")
```

Comment:
I think for whitelist extensions it should NOT set `nosniff`? @sfc-gh-hpathak ?

TY for adding this Karen!

Reply:
@sfc-gh-jcarroll based on  comment https://github.com/snowflakedb/security_reviews/pull/2712#issuecomment-1343172550 we should add `X-Content-Type-Options` irrespective of the file extension